### PR TITLE
✨ Mobile | Prize redemption

### LINKED ISSUE #736: ✨ Mobile | Reward redemption
Cc: @tkapa @matt-goldman @AntPolkanov @kristenhu

Hi team
  
As per our conversation with Camilla, the following should be done to handle reward redemptions in the app:

### Tasks

- [ ] In the admin portal, allow for rewards to be enabled and disabled
- [ ] In the mobile app, capture the user's address on redemption
    - [ ] only allow addresses from Australia
- [ ] Send an email to marketing with this information to be processed

Thanks!

### PR DESCRIPTION
**Pull request by zacharykeeping**
> 1. What triggered this change? (PBI link, Email Subject, conversation + reason, etc)

Part of #560 #736 
Closes #736 

> 2. What was changed?

1. Add basic form validation on address input
2. Hooks up a basic redemption flow (untested)

Still to do:

1. Create the redemption success view when an item is redeemed
2. Handle the response after redemption
3. Handle errors
4. Test that this even works

> 3. Did you do pair or mob programming?

@zacharykeeping @matt-goldman and @tkapa all worked on this

Done video: https://youtu.be/awkWPhGkiPA

**Note:** Video needs a v2 before we send it live but at least demonstrates the functionality

### CODE REVIEW

File: src/Common/DTOs/AddressTypes/Address.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+namespace SSW.Rewards.Shared.DTOs.AddressTypes;
+
+public class Address
+{
+    public string streetNumber { get; set; }
```
Comment (tkapa) [inline src/Common/DTOs/AddressTypes/Address.cs:5]: We have so many of these warnings, is there a way that we can ignore or fix these?

File: src/Common/DTOs/AddressTypes/Address.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    public string streetName { get; set; }
```
Comment (tkapa) [inline src/Common/DTOs/AddressTypes/Address.cs:6]: Should these have proper casing? Or is this being done to bind to JSON? (I think we can use an attribute if that's the case)

Up to you.

```suggestion
    [JsonPropertyName("streetName")]
    public string StreetName { get; set; }
```

File: src/Common/DTOs/AddressTypes/Root.cs [hunk 3]
```diff
@@ -0,0 +1,6 @@
+namespace SSW.Rewards.Shared.DTOs.AddressTypes;
+
+public class Root
+{
+    public List<Result> results { get; set; }
```
Comment (tkapa) [inline src/Common/DTOs/AddressTypes/Root.cs:5]: Same annoying warning as above

File: src/WebAPI/appsettings.json [hunk 4]
```diff
@@ -31,5 +31,8 @@
     "ClientId": "<Add to secrets>",
     "ClientSecret": "<Add to secrets>",
     "TenantId": "<Add to secrets>"
+  },
+  "AzureMaps": {
+    "Key": "<Add to secrets>"
```
Comment (tkapa) [inline src/WebAPI/appsettings.json:36]: Can we update our `settings.json` in Keeper so that everyone can get this functionality locally?

**Review by tkapa — CHANGES_REQUESTED** [review]
Loved the video :) 

Some feedback provided.

File: src/WebAPI/appsettings.json (hunk 4 above)
Comment (matt-goldman) [reply src/WebAPI/appsettings.json:36]: This is a good catch, as it reminded me we need the infra for this as well. I've added #762 to cover this which will also include updating Keeper.

I think the tasks in there should be addressed in this PR. However:

1. That would make an already relatively large/complex PR even bigger
2. We will not get that done in time for DDD

As per our conversation on Teams, we will deploy this as is now and I will ClickOps my key into Key Vault for now. We can address the remaining tasks in #762 in a future PR.

File: src/Common/DTOs/AddressTypes/Address.cs (hunk 1 above)
Comment (matt-goldman) [reply src/Common/DTOs/AddressTypes/Address.cs:5]: As per our conversation, we will address this separately. Task captured in #763.

File: src/Common/DTOs/AddressTypes/Address.cs (hunk 2 above)
Comment (matt-goldman) [reply src/Common/DTOs/AddressTypes/Address.cs:6]: As per our conversation, we will address this separately. Task captured in #763.

File: src/Common/DTOs/AddressTypes/Root.cs (hunk 3 above)
Comment (matt-goldman) [reply src/Common/DTOs/AddressTypes/Root.cs:5]: As per our conversation, we will address this separately. Task captured in #763.

**Review by tkapa — APPROVED** [review]

**Pull request merged**